### PR TITLE
Fix 18271: Circular imports on AllOf generation with REFACTOR_ALLOF_WITH_PROPERTIES_ONLY=True 

### PR DESCRIPTION
**Pull request by zishunwei**
Fix issue: https://github.com/OpenAPITools/openapi-generator/issues/18271

When enabling REFACTOR_ALLOF_WITH_PROPERTIES_ONLY=True for a AllOf generation, the circular imports error occurs if two schemas refer to each other (like #/components/schemas/CircularAllOfRef and #/components/schemas/SecondCircularAllOfRef in spec added in test https://github.com/OpenAPITools/openapi-generator/pull/18272/files#diff-4f932bfd632d4afa271a4f3cc4d10554b4d08e8767b84aa77e50cc90c699ab3cR2709)

The root cause is it adds the import statement for the models in `allVars` and will be generated at the top of model file, the `allVars` will include all models referred in properties if enabling REFACTOR_ALLOF_WITH_PROPERTIES_ONLY=True  https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java#L934, causing the circular imports error.

Since the import statements for these models are expected to be put in `postponedModelImports` and used by the mustache template, we can fix it by removing the items of `postponedModelImports` in `modelImports` if existed. This fix will not impacts on other cases

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cbornet (2017/09) @tomplus (2018/10) @krjakbrjak (2023/02) @fa0311 (2023/10) @multani (2023/10)

### CODE REVIEW

**Comment by zishunwei** [comment]
Hi @fa0311 , could you please have a review if you've time since it might be based on your fix for Circular imports on inherited discriminators https://github.com/OpenAPITools/openapi-generator/pull/17886 before, thanks a lot! cc: @wing328

File: modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml [hunk 1]
```diff
@@ -2711,3 +2711,26 @@ components:
...
+        circularAllOfRef:
+          items:
+            $ref: '#/components/schemas/CircularAllOfRef'
+          type: array
+      type: object
+    AllOfSuperModel:
+      properties:
+        _name:
+          type: string
+      type: object
```
Comment (wing328) [inline modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml:2736]: for the tests, can you please put properties under alLOf instead as the python client tests do not have any normalizer rule enabled?

Comment (wing328) [reply modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml:2736]: can you also add a test in https://github.com/OpenAPITools/openapi-generator/blob/99ca559fd25d8ba20a472a66435e083583bdee70/samples/openapi3/client/petstore/python/tests/test_model.py by actually importing the model and use it so as to ensure there will be no exception when using the model?

Comment (zishunwei) [reply modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml:2736]: Thanks @wing328 , all comments addressed.

**Comment by wing328** [comment]
lgtm. thanks for the fix

**Comment by zishunwei** [comment]
Hi @wing328 , looks my commits are not counted in https://github.com/OpenAPITools/openapi-generator/graphs/contributors although I have 2 PRs merged, any ideas? thanks

**Comment by wing328** [comment]
you need to have at least 15 commits in order to be listed in https://github.com/OpenAPITools/openapi-generator/graphs/contributors

**Pull request merged**